src/mscorlib/src/System/Reflection/Emit/CustomAttributeBuilder.cs

```diff
@@ -70,8 +70,14 @@ public CustomAttributeBuilder(ConstructorInfo con, Object[] constructorArgs)
         // Check that a type is suitable for use in a custom attribute.
         private bool ValidateType(Type t)
         {
-            if (t.IsPrimitive || t == typeof(String) || t == typeof(Type))
+            if (t.IsPrimitive)
+            {
+                return t != typeof(IntPtr) && t != typeof(UIntPtr);
```

Comment:
While this is the most surgical fix, it might be more robust to combine the IsPrimitive and IsEnum arms into a single unified typecode check (using the switch statement that's current in the IsEnum case). They way, they'll stay consistent and this IntPtr/UIntPtr bug wouldn't have occurred in the first place.